### PR TITLE
Add containerization for samples w/ container documentation

### PR DESCRIPTION
**Pull request by devkelley**
This PR adds the ability to containerize the Ibeji Samples:
- Add dockerfiles for providers and consumers
- Add mechanism to override configuration for all containerized services
- Document how to containerize the samples
- Add README's in the container folders and removed container documentation from main readme
- Add DNS alias code to samples folder

### CODE REVIEW

File: samples/common/src/utils.rs [hunk 1]
```diff
@@ -205,7 +234,40 @@ pub async fn retrieve_invehicle_digital_twin_uri(
         }
     };
 
-    Ok(result)
+    get_uri(&result).map_err(|err| {
+        format!("Failed to retrieve the in-vehicle digital twin service's URI due to error: {err}")
```
Comment (ashbeitz) [inline samples/common/src/utils.rs:238]: Are you sure that you need the .to_string() call?  See: https://doc.rust-lang.org/std/macro.format.html

**Review by ashbeitz — COMMENTED** [review]
Just one comment.

File: README.md [hunk 2]
```diff
@@ -75,7 +72,8 @@ sudo apt install -y protobuf-compiler
 
 ### <a name="install-sdl2-library">Install SDL2 library</a>
 
-You will need to install the libsdl2-dev library. This can be done by executing:
+You will need to install the libsdl2-dev library. This is used by the streaming sample to view
```
[the remark this answers is not in the document]
Comment (devkelley) [reply README.md:75]: I believe it is currently needed if you want to build the workspace, ideally at some point this gets scoped, but for now I was just adding some more context as to why this dependency exists

File: container/README.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+#### Running in Podman with overridden configuration
+
+Follow the steps in [Running in Podman](#running-in-podman) to build the container.
+
+1. To run the container with overridden configuration, create your config file and set an
+environment variable called IBEJI_HOME to the path to the config file:
+
+    ```shell
+    export IBEJI_HOME={path to directory containing config file}
```
[the remark this answers is not in the document]
Comment (devkelley) [reply container/README.md:110]: It is supposed to be the path to the directory, ie. /sdv/ibeji-config/config.yaml then IBEJI_HOME=/sdv/ibeji-config

[the remark this answers is not in the document]
Comment (devkelley) [reply container/README.md]: Updated wording, please take another look

File: samples/common/src/utils.rs [hunk 4]
```diff
@@ -205,7 +234,40 @@ pub async fn retrieve_invehicle_digital_twin_uri(
...
+            .to_string()
+    })
+}
+
+/// If feature 'containerize' is set, will modify a localhost uri to point to container's localhost
```
[the remark this answers is not in the document]
Comment (devkelley) [reply samples/common/src/utils.rs]: Updated

File: samples/common/src/utils.rs (hunk 1 above)
Comment (devkelley) [reply samples/common/src/utils.rs:237]: good catch, removed

**Review by ashbeitz — APPROVED** [review]
Approved.

**Review by jorchiu — APPROVED** [review]

**Review by jorchiu — DISMISSED** [review]

File: .github/workflows/rust-ci.yml [hunk 5]
```diff
@@ -20,7 +20,7 @@ jobs:
         with:
           submodules: recursive
       - name: Install packages
-        run: sudo apt-get install -y protobuf-compiler libsdl2-dev
+        run: sudo apt-get update && sudo apt-get install -y protobuf-compiler libsdl2-dev
```
Comment (ashbeitz) [inline .github/workflows/rust-ci.yml:23]: ... update -y ...

File: .github/workflows/rust-ci.yml [hunk 6]
```diff
@@ -54,7 +54,7 @@ jobs:
...
```
Comment (ashbeitz) [inline .github/workflows/rust-ci.yml:57]: ... update -y ...

**Review by ashbeitz — COMMENTED** [review]
Some comments.  Please take a look.

**Review by ashbeitz — APPROVED** [review]
Approved

**Pull request merged**